### DISCUSSION TITLE
why my copilot chat on VS Code only can read about 35 lines with txt

### DISCUSSION THREAD

**Discussion by Fuui944x** [community/community#147840] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I'm trying to use copilot chat to try to analysis logs which is about system error. But it seems only can read about 35 lines even I tried any ways to select whole file(e.g. select all lines then ask, use # + file name as tips showed). It is obviously that only 35 lines' log can't analysis the problem is, so it can't help me to analysis logs.
So is there some settings can let it read more lines when I ask it? Or maybe it wasn't born for this? Anyway, 35 lines quote of code is not enough as I think.
![image](https://github.com/user-attachments/assets/2a0d2384-ff98-42ed-8804-8820f08ad14c)
![image](https://github.com/user-attachments/assets/a73cb077-ec80-478e-8a65-75a531bd8d84)

**Comment by Akash1134** [comment, marked as answer]
Copilot Chat currently has a limitation on the number of lines it can process from a single file at once. This is a known constraint to ensure performance and usability. Unfortunately, there are no settings available to increase this limit.

However, here are some alternative approaches you can consider:

- Break Down Logs: Split your log file into smaller chunks of 35 lines or less and analyze each chunk separately.
- Summarize Logs: If possible, summarize the key points or errors in your logs before asking Copilot Chat for analysis.
- Use External Tools: Use external log analysis tools or scripts to pre-process and filter the logs before bringing the results back to Copilot Chat for further analysis.
